src/generator.cs

```diff
@@ -1117,6 +1127,176 @@ static string FindProtocolInterface (Type type, MemberInfo pi)
 		return "I" + type.Name;
 	}
 
+	// TODO: Alex
+	public static BindAsAttribute GetBindAsAttribute (ICustomAttributeProvider cu) => GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter);
+	public static bool HasBindAsAttribute (ICustomAttributeProvider cu) => (GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter)) != null;
+	static bool IsSetter (MethodInfo mi) => mi.IsSpecialName && mi.Name.StartsWith ("set_", StringComparison.Ordinal);
+	bool IsSmartEnum (Type type) => (type.IsValueType && type.GetFields ().Any (f => GetAttribute<FieldAttribute> (f) != null)) || Type.GetType (type.AssemblyQualifiedName.Replace (type.Name, $"{type.Name}Extensions")) != null;
+	static string GetBindAsExceptionString (string box, string retType, string containerType, string container, string memberName) => $"Could not {box} type {retType} from {containerType} {container} used on {memberName} member decorated with [BindAs].";
+
+	static Dictionary<Type,string> NSValueCreateMap = new Dictionary<Type, string> {
+		{ typeof (CGAffineTransform), "CGAffineTransform" }, { typeof (NSRange), "Range" },
+		{ typeof (CGVector), "CGVector" }, { typeof (SCNMatrix4), "SCNMatrix4" },
+		{ typeof (CLLocationCoordinate2D), "CLLocationCoordinate2D" }, { typeof (SCNVector3), "Vector" },
+		{ typeof (SCNVector4), "Vector" },
+#if XAMCORE_2_0
+		{ typeof (CGPoint), "CGPoint" }, { typeof (CGRect), "CGRect" }, { typeof (CGSize), "CGSize" },
+#endif
+#if !MONOMAC
+		{ typeof (UIEdgeInsets), "UIEdgeInsets" }, { typeof (UIOffset), "UIOffset" },
+		{ typeof (MKCoordinateSpan), "MKCoordinateSpan" },
+#endif
+#if !WATCH && !TVOS
+		{ typeof (CMTimeRange), "CMTimeRange" }, { typeof (CMTime), "CMTime" },
+		{ typeof (CMTimeMapping), "CMTimeMapping" }, { typeof (CATransform3D), "CATransform3D" },
+#endif
+	};
+
+	string GetToBindAsWrapper (MemberInformation minfo = null, ParameterInfo pi = null, PropertyInfo propInfo = null)
```

Comment:
`GetToBindAsWrapper` is never called with `propInfo`, so it looks like you can remove this parameter (and the corresponding code).